### PR TITLE
Use "npm ci" to install cspell and respect package-lock.json

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

**Review by gearama — APPROVED** [review]

**Comment by mikeharder** [comment]
If this fixes the issue, I approve, because `npm ci` is the more correct command anyway.

If this doesn't fix the issue, you might be hitting https://github.com/nodejs/node/issues/46542, which is fixed in Node 18.14.2 (npm 9.5.0).

**Comment by danieljurek** [comment]
Example spell check working (rebased on `eng/common` sync PR) https://dev.azure.com/azure-sdk/public/_build/results?buildId=2601803&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=7b4713e3-e110-5049-e231-cc8a1e664c68

File: eng/common/spelling/Invoke-Cspell.ps1 [hunk 1]
```diff
@@ -167,9 +167,9 @@ $originalLocation = Get-Location
 
 try {
   Set-Location $PackageInstallCache
-  npm install npx | Out-Null
-  npm install cspell | Out-Null
-  npm install | Out-Null
+  npm ci npx | Out-Null
+  npm ci cspell | Out-Null
```
Comment (weshaggard) [inline eng/common/spelling/Invoke-Cspell.ps1:171]: Why do we need to specifically install npx and cspell here?

Comment (mikeharder) [reply eng/common/spelling/Invoke-Cspell.ps1:171]: I would also remove the `| Out-Null`, I think it's better to keep the logs than send them to null.

File: eng/common/spelling/Invoke-Cspell.ps1 [hunk 2]
```diff
@@ -167,9 +167,9 @@ $originalLocation = Get-Location
...
+  npm ci | Out-Null
```
Comment (mikeharder) [inline eng/common/spelling/Invoke-Cspell.ps1:172]:
```suggestion
  npm ci
```

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

File: eng/common/spelling/Invoke-Cspell.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/common/spelling/Invoke-Cspell.ps1]: This script produces output which is evaluated by other code. To that end, one should run `npm ci | Write-Host` to preserve the logs but not have the additional stdout from `npm ci` interfering with upstream callers. 

Yes, I agree we should keep the logs.

**Comment by mikeharder** [comment]
LGTM

**Comment by danieljurek** [comment]
Seeing this every time I try to get sync PRs opened...

```
,##[warning]There was a failure in sending the provision message: An unexpected failure occurred while attempting to contact the remote provider.
```

**Review by weshaggard — APPROVED** [review]

**Pull request merged**